packages/muelu/doc/TutorialReadTheDocs/pages/ml_parameterlist_interpreter.rst

```diff
@@ -0,0 +1,92 @@
+============================
+ML ParameterList interpreter
+============================
+
+Backwards compatibility
+=======================
+
+ML [1]_ is the predecessor multigrid package of MueLu in Trilinos and widely used in the community for smoothed aggregation multigrid methods. ML is implemented in C and known for its good performance properties. However, the disadvantage is that ML is harder to adapt to new applications and non-standard problems. Furthermore, ML uses its own internal data structure and is somewhat limited to the use with Epetra objects only. In contrast, MueLu provides a fully flexible multigrid framework which is designed to be adapted to any kind of new application with non-standard requirements. Furthermore, it is based on Xpetra and therefore can be used both with Epetra or Tpetra. Nevertheless, it is an important point to provide some kind of backwards compatibility to allow ML users to easily migrate to MueLu (or make experiments with MueLu without having to write to much new code).
+
+In this tutorial we present the **MueLu::MLParameterListInterpreter** which provides support for the most important ML parameters to be used with MueLu.
+
+C++ part
+========
+Preparations
+------------
+In order to use MueLu (instead or aside of ML) you first have to add it to your application. Please refer to the MueLu user guide for information about compilation and linking (see [2]_). Basically, if your application is already working with ML you should only need to compile and install MueLu and make sure that the MueLu libraries are found by the linker.
+
+C++ interface
+-------------
+In the following we assume that the linear operator :math:`A` is available as **RCP<Xpetra::Matrix> A**.
+
+Then we create a parameter list and fill it with ML parameters. Please refer to the ML guide [1]_ for a complete list of available parameters.
+
+.. literalinclude:: ../../../test/tutorial/MLParameterList.cpp
+  :language: cpp
+  :start-after: ParameterList begin
+  :end-before: ParameterList end
+
+.. note::
+	Be aware that the MLParameterListInterpreter does not support all ML parameters but only the most important ones (e.g., smoothers, transfer operators, rebalancing, ...). There is, e.g., no support for the Maxwell specific enhancements in ML.
```

Comment:
The statement about Maxwell is wrong.

Reply:
@cgcgcg What would be correct?

Reply:
For both Maxwell1 and RefMaxwell, you can take an ML parameterlist, specify "parameterlist: syntax" = "ml" and stuff it in the usual constructor.